### PR TITLE
Bugfix/cpreq fixes

### PR DESCRIPTION
**Pull request by RoshanShrestha-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

• The EVS_fix directory should be symlinked locally as "fix"
• Driver scripts should point all exports of "HOMEevs" to the EVS test directory
• All stats and plots driver scripts should change COMIN to the parallel directory (/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver_2d)

• The precip and snowfall jobs should use qsub -v vhr=00 to submit jobs

• From the EVS testing directory, the following scripts should be run 
- dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_nam_precip_stats.sh
- dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_nam_snowfall_stats.sh
- dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_nam_grid2obs_stats.sh
- dev/drivers/scripts/plots/mesoscale/jevs_mesoscale_headline_plots.sh
- dev/drivers/scripts/plots/mesoscale/jevs_mesoscale_grid2obs_plots.sh
(Rest of the mesoscale scripts can be tested too, however, the above jobs should be able to catch problems if there is any.)

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

File: scripts/stats/mesoscale/exevs_mesoscale_nam_grid2obs_stats.sh [hunk 1]
```diff
@@ -362,7 +362,10 @@ echo "*****************************"
         MODEL_DIR=$(echo ${MODEL_DIR_PATH##*/})
         mkdir -p $COMOUT/$MODEL_DIR
         for FILE in $MODEL_DIR_PATH/*; do
-           cpreq -v $FILE $COMOUT/$MODEL_DIR/.
+           if [ -s $FILE ]; then
+              cp -v $FILE $COMOUT/$MODEL_DIR/.
+	   fi
+           # cpreq -v $FILE $COMOUT/$MODEL_DIR/.
```
Comment (AliciaBentley-NOAA) [inline scripts/stats/mesoscale/exevs_mesoscale_nam_grid2obs_stats.sh]: @RoshanShrestha-NOAA Can you please remove this commented out "cpreq" line? Thank you!
`# cpreq -v $FILE $COMOUT/$MODEL_DIR/.`

Comment (AliciaBentley-NOAA) [reply scripts/stats/mesoscale/exevs_mesoscale_nam_grid2obs_stats.sh]: In fact, if you could remove all commented out "cpreq" lines from this PR (I see several now), that would be great. Thank you!

@RoshanShrestha-NOAA @PerryShafran-NOAA

**Comment by AliciaBentley-NOAA** [comment]
I've asked @RoshanShrestha-NOAA to remove the commented out "cpreq" lines in this PR before we merge it. I don't think those lines will impact testing, if @PerryShafran-NOAA wanted to start testing this PR this morning. 

Once tested, I approve this PR for merge. CC: @ShelleyMelchior-NOAA

File: scripts/stats/mesoscale/exevs_mesoscale_nam_grid2obs_stats.sh (hunk 1 above)
Comment (RoshanShrestha-NOAA) [reply scripts/stats/mesoscale/exevs_mesoscale_nam_grid2obs_stats.sh]: I will remove them asap

**Comment by RoshanShrestha-NOAA** [comment]
Note: I wrote "Wei-#86" referring to Item #86 from Wei, but github changed it to "Wei-NOAA-EMC#86" linking to PR#86. Don't get confused over it. Sorry for the inconvenience.

**Comment by PerryShafran-NOAA** [comment]
@RoshanShrestha-NOAA Do you have a testing strategy that you'd like me to undertake?  Or basically just test every script in the mesoscale component?  There's no instructions on how to test this so I'm curious what you would like me to test.

**Comment by RoshanShrestha-NOAA** [comment]
@PerryShafran-NOAA , I will add some instructions. Thanks.

> @RoshanShrestha-NOAA Do you have a testing strategy that you'd like me to undertake? Or basically just test every script in the mesoscale component? There's no instructions on how to test this so I'm curious what you would like me to test.

**Comment by RoshanShrestha-NOAA** [comment]
@PerryShafran-NOAA I have added instructions above. This is just a copy of it.

• The EVS_fix directory should be symlinked locally as "fix"
• Driver scripts should point all exports of "HOMEevs" to the EVS test directory
• All stats and plots driver scripts should change COMIN to the parallel directory (/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver_2d)

• The precip and snowfall jobs should use qsub -v vhr=00 to submit jobs

• From the EVS testing directory, the following scripts should be run 
- dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_nam_precip_stats.sh
- dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_nam_snowfall_stats.sh
- dev/drivers/scripts/stats/mesoscale/jevs_mesoscale_nam_grid2obs_stats.sh
- dev/drivers/scripts/plots/mesoscale/jevs_mesoscale_headline_plots.sh
- dev/drivers/scripts/plots/mesoscale/jevs_mesoscale_grid2obs_plots.sh
(Rest of the mesoscale scripts can be tested too, however, the above jobs should be able to catch problems if there is any.)

Thanks.



> @PerryShafran-NOAA , I will add some instructions. Thanks.
> 
> > @RoshanShrestha-NOAA Do you have a testing strategy that you'd like me to undertake? Or basically just test every script in the mesoscale component? There's no instructions on how to test this so I'm curious what you would like me to test.

**Comment by PerryShafran-NOAA** [comment]
@RoshanShrestha-NOAA Thank you!  I will begin testing.

**Comment by PerryShafran-NOAA** [comment]
@RoshanShrestha-NOAA Testing in the stats directory is underway.  You can follow along with all the .o files here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr436test/EVS/dev/drivers/scripts/stats/mesoscale

**Comment by PerryShafran-NOAA** [comment]
@RoshanShrestha-NOAA The precip and snowfall jobs have finished.  Would you please take a look at the output?

The .o files are listed above.
The output is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/mesoscale/atmos.20240204/nam

**Comment by RoshanShrestha-NOAA** [comment]
@PerryShafran-NOAA  I looked at them, all look fine. I did not find any problems.

> @RoshanShrestha-NOAA The precip and snowfall jobs have finished. Would you please take a look at the output?
> 
> The .o files are listed above. The output is here: /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/mesoscale/atmos.20240204/nam

**Comment by RoshanShrestha-NOAA** [comment]
@PerryShafran-NOAA The nam grid2obs jobs are complete, and everything looks good.

**Comment by PerryShafran-NOAA** [comment]
@RoshanShrestha-NOAA Plot jobs are underway.  You can follow the .o files along here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr436test/EVS/dev/drivers/scripts/plots/mesoscale

**Comment by PerryShafran-NOAA** [comment]
@RoshanShrestha-NOAA Please do a sync fork.

**Comment by RoshanShrestha-NOAA** [comment]
@PerryShafran-NOAA the plot jobs are complete. Everything looks good. 

> @RoshanShrestha-NOAA Plot jobs are underway. You can follow the .o files along here: /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr436test/EVS/dev/drivers/scripts/plots/mesoscale

**Review by PerryShafran-NOAA — APPROVED** [review]
Code works as expected.  Approved for merge.

**Pull request merged**